internal/errcheck/errcheck.go

```diff
@@ -220,12 +219,31 @@ func (v *visitor) ignoreCall(call *ast.CallExpr) bool {
 			if re, ok := v.ignore[pkg.Path()]; ok {
 				return re.MatchString(id.Name)
 			}
+
+			// if current package being considered is vendored, check to see if it should be ignored based
+			// on the unvendored path.
+			if nonVendoredPkg, ok := nonVendoredPkgPath(pkg.Path()); ok {
+				if re, ok := v.ignore[nonVendoredPkg]; ok {
+					return re.MatchString(id.Name)
+				}
+			}
 		}
 	}
 
 	return false
 }
 
+// nonVendoredPkgPath returns the unvendored version of the provided package path (or returns the provided path if it
+// does not represent a vendored path). The second return value is true if the provided package was vendored, false
+// otherwise.
+func nonVendoredPkgPath(pkgPath string) (string, bool) {
+	lastVendorIndex := strings.LastIndex(pkgPath, "/vendor/")
```

Comment:
I think the path-based approach here is reasonable given that core Go code does something similar: https://github.com/golang/go/blob/81dfcba331f43bd14c8933eca83c433e53cb7b55/src/cmd/go/build.go#L2324

The other approach would be to iterate over all of the `v.ignore` keys and try to import them and see if they resolve to a vendored package, but this requires more information (need a source directory from which to try to resolve import), so I think this approach is reasonable. However, would be happy to hear other feedback.